src/gitingest/clone.py

```
@@ -103,42 +100,65 @@ async def clone_repo(config: CloneConfig) -> tuple[bytes, bytes]:
-        return await run_git_command(*clone_cmd)
-
-    except (RuntimeError, asyncio.TimeoutError, AsyncTimeoutError):
-        raise  # Re-raise the exception
+        if query["commit"]:
+            proc = await asyncio.create_subprocess_exec(
+                "git",
+                "clone",
+                "--single-branch",
+                query["url"],
+                query["local_path"],
+                stdout=asyncio.subprocess.PIPE,
+                stderr=asyncio.subprocess.PIPE,
+            )
+            stdout, stderr = await proc.communicate()
+
+            proc = await asyncio.create_subprocess_exec(
+                "git",
+                "-C",
+                query["local_path"],
+                "checkout",
+                query["branch"],
+                stdout=asyncio.subprocess.PIPE,
+                stderr=asyncio.subprocess.PIPE,
+            )
+            stdout, stderr = await proc.communicate()
+        elif (
+            query["branch"] != "main"
+            and query["branch"] != "master"
+            and query["branch"]
+        ):
+            proc = await asyncio.create_subprocess_exec(
+                "git",
+                "clone",
+                "--depth=1",
+                "--single-branch",
+                "--branch",
+                query["branch"],
+                query["url"],
+                query["local_path"],
+                stdout=asyncio.subprocess.PIPE,
+                stderr=asyncio.subprocess.PIPE,
+            )
+        else:
+            proc = await asyncio.create_subprocess_exec(
+                "git",
+                "clone",
+                "--depth=1",
+                "--single-branch",
+                query["url"],
+                query["local_path"],
+                stdout=asyncio.subprocess.PIPE,
+                stderr=asyncio.subprocess.PIPE,
+            )
+
+        stdout, stderr = await proc.communicate()
+        return stdout, stderr
+    except TimeoutError:
```

We kind of need to raise the error for the current web version to work, but I now realise that this is not ideal for the CLI